### PR TITLE
GitHub cache

### PR DESCRIPTION
**Pull request by sidharthv96**
Caches the report.json file to github action cache.

We could theoretically do this for the npm install also, but handling npm/yarn would need to be done.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -45,19 +45,5 @@ inputs:
...
-              INPUT_WORKING-DIRECTORY: ${{ inputs.working-directory }}
-              INPUT_ICONS: ${{ inputs.icons }}
-              INPUT_ANNOTATIONS: ${{ inputs.annotations }}
-              INPUT_PACKAGE-MANAGER: ${{ inputs.package-manager }}
-              INPUT_SKIP-STEP: ${{ inputs.skip-step }}
-              INPUT_CUSTOM-TITLE: ${{ inputs.custom-title }}
-              INPUT_COVERAGE-FILE: ${{ inputs.coverage-file }}
-              INPUT_BASE-COVERAGE-FILE: ${{ inputs.base-coverage-file }}
+    using: node12
+    main: 'dist/index.js'
```
Comment (sidharthv96) [inline action.yml:49]: Need to use this to get some internal ENV Vars populated which is used in the @actions/cache package.

Comment (ArtiomTr) [reply action.yml:49]: The reason why the run.sh script was previously used is described here (#122). In short, the use of this script made it possible to not fix the version of the node - user can pick it manually. 
I've checked `@actions/cache` source code, so they are using only two environment variables - `GITHUB_EVENT_NAME` and `GITHUB_REF`. So, you can try to just pass these values, as it was in old configuration:
```yml
env:
    GITHUB_EVENT_NAME: ${{ env.GITHUB_EVENT_NAME }}
    GITHUB_REF: ${{ env.GITHUB_REF }}
```

File: esbuild.cjs [hunk 2]
```diff
@@ -2,8 +2,9 @@
 const { build } = require('esbuild');
 build({
     bundle: true,
-    minify: true,
+    minify: false, // Cannot minify due to https://github.com/node-fetch/node-fetch/issues/784
```
Comment (ArtiomTr) [inline esbuild.cjs:5]: I think the [keep names](https://esbuild.github.io/api/#keep-names) option should fix this issue, without disabling minification

**Review by ArtiomTr — CHANGES_REQUESTED** [review]
Looks good, but I would like to ask you to take a look at some places

File: action.yml (hunk 1 above)
Comment (sidharthv96) [reply action.yml:49]: https://github.com/actions/toolkit/blob/ea81280a4d48fb0308d40f8f12ae00d117f8acb9/packages/cache/src/internal/cacheHttpClient.ts#L36

This is the env var that's required, and was not being populated for composite actions.

https://github.com/actions/toolkit/issues/852 

There seems to be an issue open in the repo.

File: esbuild.cjs (hunk 2 above)
Comment (sidharthv96) [reply esbuild.cjs]: Haha, I did try it yesterday but it didn't seem to work, VSCode find only found 3 instances of AbortSignal in the code with and without minify enabled, So I thought it wasn't working..

But I just tried with grep and it shows 7 vs 9 with keepNames.
Guess VSCode tricked me.

Comment (sidharthv96) [reply esbuild.cjs]: This bug was a real elusive one as the error message doesn't say much and the cache saving was working properly.

File: action.yml (hunk 1 above)
Comment (dapplion) [reply action.yml:49]: Does this work now? Are you sure the ENVs `ACTIONS_CACHE_URL` or `ACTIONS_RUNTIME_URL` are injected? I would suggest adding a console.log() to check it

Comment (sidharthv96) [reply action.yml:49]: I logged process.env to check.
When running as composite action, even after injecting to env as @ArtiomTr suggested, I wasn't seeing the envs.
Only making it a node function worked. 

But tying it to node12 is a step back from current functionality.

Comment (sidharthv96) [reply action.yml:49]: Working logs : https://github.com/sidharthv96/jest-coverage-report-action-test/pull/2/checks

Comment (ArtiomTr) [reply action.yml:49]: @sidharthv96 I need time to dig deeper into the problem and maybe find a solution. I hope you don't mind if this PR is delayed for a little.

Comment (sidharthv96) [reply action.yml:49]: No problem at all, this was just one feature I thought might help speed up the test runs when writing the comments PR.

Nice to have, not need to have. Especially if it breaks compatability.

Comment (sidharthv96) [reply action.yml:49]: One extremely hacky way we could enable someone who really needs to speed up tests is.

Something in documentation as an example.

I really don't think it's a good idea, but just throwing it out there 😂

- sh command to get base ref hash if it exists
- Use GitHub cache action with report path and ref as key
- Run our action as composite itself
- check if report exists, use it if does as base branch report.
- switch to head, run tests, compare.
- write **base report** back to path 
- GH cache action will store it. 

Not really sure if it's worth the effort though.

**Comment by ArtiomTr** [comment]
@sidharthv96 could you please change the target branch of this PR? Github has automatically closed this PR and won't let me reopen it because the `Refactor` branch has been deleted.

**Pull request closed without merging**